lib/Push.php

```diff
@@ -154,21 +202,38 @@ public function pushToDevice(int $id, INotification $notification, ?OutputInterf
 			return;
 		}
 
+		if ($this->deferPreparing) {
+			$this->notificationsToPush[$id] = clone $notification;
```

Comment:
I guess that shallow clones of a notification will work as expected, but shallow clones of objects defined elsewhere are a bit scary :-)

Reply:
There are no sub-objects, but notification objects are reused outside to loop for multiple users e.g. so we need to clone them

Reply:
There can be sub-objects (for example the validator or the actions), but I guess there would be no problem anyway. My concern was more regarding future sub-objects that could be added in the future and be problematic, but I guess we can worry about that when/if that time comes.